src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -320,6 +349,19 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig, protected
     } yield accessConfig.getNatIP
   }
 
+  private def getInitBucketName(cluster: GoogleCluster): Option[String] = {
+    def parseBucketName(bucketPath: String): Option[String] = {
+      Try(Option(new URI(bucketPath).getHost)).toOption.flatten
```

Comment:
@jmthibault79 pointed out that `URI` should not be used to parse bucket paths because GCS buckets can have names that aren't valid URI hosts. E.g.:
```
scala> new java.net.URI("gs://a_bucket_with_underscores/foo")
res4: java.net.URI = gs://a_bucket_with_underscores/foo

scala> res4.getHost
res5: String = null
```
I'll find another solution here.